spyder/plugins/variableexplorer/widgets/objectexplorer/objectexplorer.py

```diff
@@ -205,122 +179,125 @@ def set_value(self, obj):
             old_obj_tree.deleteLater()
         else:
             self.central_splitter.insertWidget(0, self.obj_tree)
-
+            self.central_splitter.setCollapsible(0, False)
+            self.central_splitter.setCollapsible(1, True)
+            self.central_splitter.setSizes([500, 320])
 
     def _make_show_column_function(self, column_idx):
         """Creates a function that shows or hides a column."""
         show_column = lambda checked: self.obj_tree.setColumnHidden(
             column_idx, not checked)
         return show_column
 
-    def _setup_actions(self):
-        """Creates the main window actions."""
+    def _setup_toolbar(self, show_callable_attributes=False,
+                       show_special_attributes=False):
+        """
+        Sets up the toolbar and the actions in it.
+        """
+        def do_nothing():
+            # .create_action() needs a toggled= parameter, but we can only
+            # set it later in the set_value method, so we use this function as
+            # a placeholder here.
+            pass
+
+        self.toolbar = self.create_toolbar(
+            'Object explorer toolbar', register=False
+        )
+
         # Show/hide callable objects
-        self.toggle_show_callable_action = QAction(
-            _("Show callable attributes"),
-            self,
-            checkable=True,
-            shortcut=QKeySequence("Alt+C"),
-            statusTip=_("Shows/hides attributes that are callable "
-                        "(functions, methods, etc)")
+        self.toggle_show_callable_action = self.create_action(
+            name='Show callable attributes',
+            text=_("Show callable attributes"),
+            icon=ima.icon("class"),
+            tip=_("Shows/hides attributes that are callable "
+                  "(functions, methods etc)"),
+            toggled=self._show_callable_attributes,
+            option='show_callable_attributes',
+            register_action=False
         )
+        self.toolbar.add_item(self.toggle_show_callable_action)
```

Comment:
I think it be better if you introduce all items after creating them first (i.e. at the end of this method) and by using `add_item_to_toolbar`.